### ISSUE TITLE
Doesn’t run on wine

### ISSUE THREAD

**Issue by andrewshadura** [msys2/msys2.github.io#178]
I’m trying to run msys2 on Wine:
```
C:\msys64>usr/bin/bash
Cygwin WARNING:
  Couldn't compute FAST_CWD pointer.  This typically occurs if you're using
  an older Cygwin version on a newer Windows.  Please update to the latest
  available Cygwin version from https://cygwin.com/.  If the problem persists,
  please see https://cygwin.com/problems.html


C:\msys64>
```
The same error appears with many other msys2 binaries.

**Comment by lazka** [comment]
I took the opportunity and wrote a FAQ entry for this: https://www.msys2.org/docs/faq/#how-can-i-get-msys2-running-under-wine

tl;dr: you sadly can't, at least right now.

**Comment by jeremyd2019** [comment]
Also, the `FAST_CWD` warning is a red herring.  It's one more thing that doesn't work on wine, but it isn't fatal.

**Comment by andrewshadura** [comment]
Yes, I realised it’s a warning when I was posting it, but it failed anyway so I reported it. Sad to learn this has been a known issue for years with no solution.

**Comment by ZachBacon** [comment]
The issue in this case sadly falls on the wine team to solve.

**Comment by BenjaminRi** [comment]
> The issue in this case sadly falls on the wine team to solve.

This is not a Wine bug. See: https://bugs.winehq.org/show_bug.cgi?id=40528

> NOTOURBUG makes sense to me. I don't think the Wine Windows API implementation can be reasonably expected to offer random internal windows structures for direct manipulation.

Cygwin meddles with private, internal Windows structures that are not present under Wine.

**Comment by Biswa96** [comment]
> This is not a Wine bug. See: https://bugs.winehq.org/show_bug.cgi?id=40528

Yes, you're right. To be specific, the FAST_CWD error happens because cygwin and msys2 poke into specific machine instruction to get hidden string in ntdll.dll. See `find_fast_cwd_pointer()` in https://github.com/cygwin/cygwin/blob/main/winsup/cygwin/path.cc

But there are some issue in wine side also which are related to cygwin. For example, https://bugs.winehq.org/show_bug.cgi?id=50771 and [others](https://bugs.winehq.org/buglist.cgi?quicksearch=cygwin).

tl;dr Need more hard work... 💪

**Comment by Kreijstal** [comment]
couldn't you just patch that cygwin code that meddles with the system with a wine specific version
